app/src/lib/api.ts

```diff
@@ -500,31 +525,26 @@ export class API {
    * pages when available, buffers all items and returns them in
    * one array when done.
    */
-  private async fetchAll<T>(path: string): Promise<ReadonlyArray<T>> {
+  private async fetchAll<T>(path: string, options?: IFetchAllOptions<T>) {
     const buf = new Array<T>()
+    const opts: IFetchAllOptions<T> = { perPage: 100, ...options }
+    const params = { per_page: `${opts.perPage}` }
 
-    const params = {
-      per_page: '100',
-    }
     let nextPath: string | null = urlWithQueryString(path, params)
-
     do {
       const response = await this.request('GET', nextPath)
-      if (response.status === HttpStatusCode.NotFound) {
-        log.warn(`fetchAll: '${path}' returned a 404`)
-        return []
-      }
-      if (response.status === HttpStatusCode.NotModified) {
-        log.warn(`fetchAll: '${path}' returned a 304`)
-        return []
+      if (!response.ok) {
+        log.warn(`fetchAll: '${path}' returned a ${response.status}`)
+        return buf
       }
 
       const items = await parsedResponse<ReadonlyArray<T>>(response)
       if (items) {
         buf.push(...items)
       }
+
       nextPath = getNextPagePath(response)
-    } while (nextPath)
+    } while (nextPath && opts.continue && !opts.continue(buf))
```

Comment:
Is this check right? I feel if `opts.continue` is undefined this will evaluate to falsy rather than continue on (which was the previous behaviour):

<img width="372" src="https://user-images.githubusercontent.com/359239/55983083-4aa21080-5c68-11e9-8af2-03829fe51442.png">

Reply:
It certainly isn't. My bad, I converted this from being an inline if statement with a break and didn't invert the logic. Great catch!